Dockerfile

```
@@ -1,42 +1,35 @@
-
-# Set the working directory
+# Skip tests for faster CI builds (optional)
+RUN mvn clean install -DskipTests
+
+### ----------- Stage 2: Run with Lightweight JRE -----------
+FROM eclipse-temurin:21-jre-jammy
+
 WORKDIR /app
+
+# Install tini for better process management (optional but useful)
+RUN apt-get update && apt-get install -y tini && apt-get clean
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)